### PR TITLE
feat: modularity, dinar-ephemeral-alpha

### PR DESCRIPTION
**Pull request by tupakkatapa**
### modularity
I have separated the commonly used sections from `hosts/ponkila-ephemeral-beta/default.nix` to the `modules` directory. This allows other host configurations to access these settings. Variables, such as SSH authorized keys, are still managed from the host-specific configuration through mkOption.

### dinar-ephemeral-alpha + init-qemu.sh
I have added a new work-in-progress host called dinar-ephemeral-alpha. This host is going to function in a virtual environment, so I have included a script for converting our traditional kexecTree output format to a qcow2 image.

### CODE REVIEW

**Comment by jhvst** [comment]
LGTM, deployed the updates on `ponkila-ephemeral-beta`, worked fine. Still, `sops` is failing, so the following commands have to be done post-boot:

1.  `systemctl --user restart sops-nix.service`
2. `sudo systemctl restart mev-boost.service erigon.service lighthouse.service`

This is because the sops-nix service attempts to run early into the boot process, and the secrets path are not mounted yet. We need to see if the `sops-nix` service can be altered to run only after the systemd mounts are done, but insofar even `lib.makeForce` option to override the `after` segment of the `sops-nix` user service does not seem to take effect.

Further, the setup currently assumes that the following mount points (all formatted as `btrfs`) are true:

- `/var/mnt/erigon` for erigon
- `/var/mnt/lighthouse` for lighthouse
- `/var/mnt/secrets` for ssh and sops

Additionally, erigon and lighthouse paths expect the `jwt.hex` to exist in their mount points. This should be later migrated to be a `sops` secret as well.

Before merging, let's ensure that the generation of `qcow` format from the initrd and kernel works, and can be deployed on proxmox.

**Review by jhvst — APPROVED** [review]
With the addition of KVM kernel modules, QEMU is booting fine. Needs a final check to see if this is enough for Proxmox too.

**Comment by tupakkatapa** [comment]
Reverted back to using `nix-community/ethereum.nix` with the latest update of Lighthouse from v4.0.1 to v4.0.2-rc.0, as the pull request was successfully merged.

https://github.com/nix-community/ethereum.nix/pull/217

File: init-qemu.sh
Comment (jhvst) [inline init-qemu.sh:1]: This file should be removed, as we should favor using ISO files instead for more versatile boot options (USB, and so on).

**Comment by tupakkatapa** [comment]
Resolved issues in the reviews identified by Juuso.

Created a custom format for generating ISO files. This format is derived from the [iso.nix](https://github.com/nix-community/nixos-generators/blob/master/formats/iso.nix) format of nixos-generators, with copytoram enabled and grub timeout set to 1s.

**Pull request merged**